engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/syntax/HostValueToEnsoNode.java

```diff
@@ -0,0 +1,55 @@
+package org.enso.interpreter.node.expression.builtin.interop.syntax;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.ReportPolymorphism;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+
+/**
+ * Converts a value returned by a polyglot call back to a value that can be further used within Enso
+ * programs.
+ */
```

Comment:
If we already have such mechanism, maybe it is worth revisiting wrapping the big-integers into the `EnsoBigInteger` wrapper and instead using `BigIntegers` but converting them here to the underlying-Enso-compatible type?

Reply:
What we get here would not be a `BigInteger`, but a polyglot wrapper (a host object). There are ways to get a `BigInteger` back, but I'm weary of the consequences – the `BigInteger` may be coming from a different classloader, it may be a subclass. It will also never work with other languages. So I refuse, until (if?) big integer becomes a legit part of the interop library.